### ISSUE TITLE
HttpListenerPrefixCollection.CopyTo(null, 0) throws an NRE

### ISSUE THREAD

**Issue by hughbe** [dotnet/corefx#19438]
The following test demonstrates this bug that is present in netfx and the Windows implementation of HttpListener. Note that the managed implementation does not have this bug.

Although HttpListener is considered a legacy/compat API, @stephentoub mentioned in #19217:

> We don't need bug-for-bug compatibility. It's a bug in desktop that it throws an NRE; we don't need to go out of our way to mimic that.

So I'm tracking this with an issue

```cs
[Fact]
public void CopyTo_NullArray_ThrowsNullReferenceException()
{
    var listener = new HttpListener();
    Assert.Throws<NullReferenceException>(() => listener.Prefixes.CopyTo((Array)null, 0));
    Assert.Throws<NullReferenceException>(() => listener.Prefixes.CopyTo(null, 0));
}
```

**Comment by karelz** [comment]
So the difference is HttpListener on Linux & Mac vs. Windows & .NET Framework, right?

Is it something we want to fix? In Windows Core?

**Comment by hughbe** [comment]
Correct. I think @stephentoub hinted this should be fixed in #19127.

This was the issue I mentioned on twitter btw @karelz: it's unclear if we should fix In in corefx. Most people don't rely on catching NREs so a fix in .Net core is likely to help incentivise a netfx fix for this bug.

**Comment by karelz** [comment]
Thanks for filing it @hughbe! It is always good to have technical discussion before we make decisions ;). Here's my take:

Fix in NetFX could be compat problem. So unless we have real-world customers hitting the bug, it is too much risk to fix it there IMO.
Note that .NET Framework is **pushed** everywhere - it just takes one populr/important app depending on the old behavior and you are doing disservice to the end users and developers. From that perspective, we should not fix things in .NET Framework, unless, either:
1. It is (a) hurting customers (we have reports) & (b) it is not too risky/breaking to fix it, ir
2. The behavior is (a) clearly bogus & (b) people are highly unlikely to depend on it (low compat risk) & (c) we believe developers will use it in future, without reasonable workaround available.

This issue (on .NET Framework) falls doesn't satisfy (1a). In (2) - I am not certain about (2b) and (2c) is unlikely given the API exists for quite some time.

Given that HttpListener is legacy API in .NET Core, I'd suggest to close it, unless it is super-simple to fix and someone is motivated to do it ;-).
In other words: I would not block a community PR, but I would not also ask anyone (even in community) to fix it, as it is IMO not worth it ... almost any bug in CoreFX repo is more impactful and higher priority than this one ;).
